### ISSUE TITLE
Run with GPUs

### ISSUE THREAD

**Issue by kinhunt** [ardiya/HashtagPrediction#4]
I have a machine with 2 GPUs. How to use them in with your project? Sorry for this newbie question.

**Comment by ardiya** [comment]
Not sure if this will work or not since I dont have multi-GPU, but try this link. https://github.com/tensorflow/models/blob/master/slim/deployment/model_deploy.py
Based on the usage, you can use it like this:

  g = tf.Graph()
  # Set up DeploymentConfig
  **config = model_deploy.DeploymentConfig(num_clones=2, clone_on_cpu=True)**
  # Create the global step on the device storing the variables.
  **with tf.device(config.variables_device()):**
    global_step = slim.create_global_step()
  # Define the inputs
  **with tf.device(config.inputs_device()):**
    images, labels = LoadData(...)
    inputs_queue = slim.data.prefetch_queue((images, labels))
  # Define the optimizer.
  **with tf.device(config.optimizer_device()):**
    optimizer = tf.train.MomentumOptimizer(FLAGS.learning_rate, FLAGS.momentum)
  # Define the model including the loss.
  def model_fn(inputs_queue):
    images, labels = inputs_queue.dequeue()
    predictions = CreateNetwork(images)
    slim.losses.log_loss(predictions, labels)
  **model_dp = model_deploy.deploy(config, model_fn, [inputs_queue],
                                 optimizer=optimizer)**
  # Run training.
  slim.learning.train(model_dp.train_op, my_log_dir,
                      summary_op=model_dp.summary_op)

**Comment by kinhunt** [comment]
thank you

**Comment by kinhunt** [comment]
could you help me of setting up the deployment?
```
model_dp = model_deploy.deploy(config, model_fn, [inputs_queue],
optimizer=optimizer)
```
How to define the model_fn? Are inputs_queue and optimizer necessary?

**Comment by ardiya** [comment]
probably
```
def model_fn(batch_x, batch_y):
    logits = model.inference(batch_x)
    losses = model.loss(logits, batch_y)

model_dp = model_deploy.deploy(config, model_fn, [batch_x, batch_y],
                                 optimizer=tf.train.AdamOptimizer())
```
mode_fn should include inference and loss
inputs_queue is the input of the net
optimizer can be choose from https://www.tensorflow.org/versions/r0.11/api_docs/python/train.html#optimizers, it is how you update the weight through backpropagation, so it is necessary

**Comment by kinhunt** [comment]
It seems to be working. Thank you.
